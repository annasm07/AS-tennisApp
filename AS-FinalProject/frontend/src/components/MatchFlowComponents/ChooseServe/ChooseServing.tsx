import * as React from 'react';
import {Text} from 'react-native';
import {SafeAreaView} from 'react-native-safe-area-context';

export default function Stats() {
  return (
    <SafeAreaView>
      <Text>Choose serving PAGE IS WORKING</Text>
    </SafeAreaView>
  );
}
