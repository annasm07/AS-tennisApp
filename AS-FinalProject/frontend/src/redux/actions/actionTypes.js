export default {
  LOG_IN: 'LOG_IN',
  LOG_IN_ERROR: 'LOG_IN_ERROR',
  SIGN_UP: 'LOG_IN',
  SIGN_UP_ERROR: 'LOG_IN_ERROR',
  GET_PLAYER: 'GET_PLAYER',
  GET_PLAYER_ERROR: 'GET_PLAYER_ERROR',
  GET_PLAYERS: 'GET_PLAYERS',
  GET_PLAYERS_ERROR: 'GET_PLAYERS_ERROR',
  CREATE_MATCH: 'CREATE_MATCH',
  CREATE_MATCH_ERROR: 'CREATE_MATCH_ERROR',
  UPDATE_POINTS: 'UPDATE_POINTS',
  UPDATE_POINTS_ERROR: 'UPDATE_POINTS_ERROR',
  UPDATE_GAMES: 'UPDATE_GAMES',
  UPDATE_GAMES_ERROR: 'UPDATE_GAMES_ERROR',
  END_GAME: 'END_GAME',
  END_GAME_ERROR: 'END_GAME_ERROR',
  END_SET: 'END_SET',
  END_SET_ERROR: 'END_SET_ERROR',
  END_MATCH: 'END_MATCH',
  END_MATCH_ERROR: 'END_MATCH_ERROR',
};
