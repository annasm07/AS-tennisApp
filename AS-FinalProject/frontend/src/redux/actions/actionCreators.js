import axios from 'axios';
import actionTypes from './actionTypes';

const LOG_IN_URL = 'http://localhost:1616/api/auth/login';
const SIGN_UP_URL = 'http://localhost:1616/api/auth/signup';
const GET_PLAYER_URL = 'http://localhost:1616/api/home';
const GET_ALL_PLAYERS = 'http://localhost:1616/api/players';
const CREATE_MATCH_URL = 'http://localhost:1616/api/match';

export function logIn(email, password) {
  return async dispatch => {
    try {
      const {data} = await axios.post(LOG_IN_URL, {email, password});
      dispatch({
        type: actionTypes.LOG_IN,
        user: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.LOG_IN_ERROR,
      });
    }
  };
}

export function signUp({name, email, password, player, playerName}) {
  return async dispatch => {
    try {
      await axios.post(SIGN_UP_URL, {
        name,
        email,
        password,
        player,
        playerName,
      });
      dispatch({
        // type: actionTypes.SIGN_UP,
        // user: data.user,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.SIGN_UP_ERROR,
      });
    }
  };
}

export function getPlayerInfo(token, playerId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${GET_PLAYER_URL}/${playerId}`, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYER,
        player: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYER_ERROR,
      });
    }
  };
}
export function getAllPlayers(token) {
  return async dispatch => {
    try {
      const {data} = await axios(GET_ALL_PLAYERS, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      dispatch({
        type: actionTypes.GET_PLAYERS,
        players: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.GET_PLAYERS_ERROR,
      });
    }
  };
}

export function newMatch(token, p1Name, p2Name) {
  return async dispatch => {
    try {
      const {data} = await axios.post(
        CREATE_MATCH_URL,
        {p1Name, p2Name},
        {
          headers: {
            Authorization: `Bearer ${token}`,
          },
        },
      );
      dispatch({
        type: actionTypes.CREATE_MATCH,
        currentMatch: data,
      });
    } catch (error) {
      console.log(error);
      dispatch({
        type: actionTypes.CREATE_MATCH_ERROR,
      });
    }
  };
}
