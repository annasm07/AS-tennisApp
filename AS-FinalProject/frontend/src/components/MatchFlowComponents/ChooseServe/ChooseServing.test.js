import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';
import ConfigureStore from 'redux-mock-store';
import * as actions from '../../../redux/actions/actionCreators';
import {Provider} from 'react-redux';
import ChooseServing from './ChooseServing';

jest.mock('../../../redux/actions/actionCreators');

describe('When invoked a Dashboard component', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {
      result: [{name: 'Anna'}, {name: 'Sala'}],
    },
  });
  let navigation;

  beforeEach(() => {
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });
  test('Should render dashboard component', () => {
    const chooseServing = render(
      <Provider store={store}>
        <ChooseServing navigation={navigation} />
      </Provider>,
    );
    expect(chooseServing).toMatchSnapshot();
  });

  test('When Player 1 is pressed, updateServer to have been called', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <ChooseServing navigation={navigation} />
      </Provider>,
    );
    const ClickPlayer1 = getByTestId('ClickPlayer1');
    fireEvent.press(ClickPlayer1);
    expect(actions.updateServer).toHaveBeenCalled();
  });

  test('When Player 2 is pressed, updateServer to have been called', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <ChooseServing navigation={navigation} />
      </Provider>,
    );
    const ClickPlayer2 = getByTestId('ClickPlayer2');
    fireEvent.press(ClickPlayer2);
    expect(actions.updateServer).toHaveBeenCalled();
  });
});

describe('When invoked a Dashboard component without result', () => {
  const mockStore = ConfigureStore();
  const store = mockStore({
    currentMatch: {},
  });
  let navigation;

  beforeEach(() => {
    jest.spyOn(actions, 'updateServer').mockReturnValueOnce({type: ''});

    navigation = {
      navigate: jest.fn(),
    };
  });
  test('When currentMatch.result does not exist, loading is rendered', () => {
    const chooseServing = render(
      <Provider store={store}>
        <ChooseServing navigation={navigation} />
      </Provider>,
    );
    expect(chooseServing).toMatchSnapshot();
  });
});
