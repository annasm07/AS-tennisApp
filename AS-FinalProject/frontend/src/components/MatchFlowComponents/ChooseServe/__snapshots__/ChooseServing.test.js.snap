// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When invoked a Dashboard component Should render dashboard component 1`] = `
<View
  style={
    Object {
      "height": 100,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 26,
        "paddingLeft": 20,
      }
    }
  >
    Who is Serving?
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      testID="ClickPlayer1"
    >
      Anna
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "margin": 15,
        "marginBottom": 5,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      testID="ClickPlayer2"
    >
      Sala
    </Text>
  </View>
</View>
`;

exports[`When invoked a Dashboard component without result When currentMatch.result does not exist, loading is rendered 1`] = `
<Text>
  ...loading...
</Text>
`;
